src/self-managed-debug/src/main.rs

```diff
@@ -0,0 +1,722 @@
+
+    let pods: Api<Pod> = Api::<Pod>::namespaced(client.clone(), namespace);
+
+    let pod_list = pods.list(&ListParams::default()).await?;
+
+    for pod in &pod_list.items {
+        if let Err(e) = async {
+            let pod_name = pod.metadata.name.clone().unwrap_or_default();
+            let file_name = format!("mz-pod-logs.{}.{}.log", namespace, pod_name);
+            let mut file = File::create(&file_name)?;
+
+            let logs = match pods
+                .logs(
+                    &pod_name,
+                    &LogParams {
+                        previous: true,
+                        timestamps: true,
+                        ..Default::default()
+                    },
+                )
+                .await
+            {
+                Ok(logs) => logs,
+                Err(_) => {
+                    // If we get a bad request error, try without the previous flag.
+                    pods.logs(
+                        &pod_name,
+                        &LogParams {
+                            timestamps: true,
+                            ..Default::default()
+                        },
+                    )
+                    .await?
+                }
+            };
+
+            for line in logs.lines() {
+                writeln!(file, "{}", line)?;
+            }
+
+            println!("Finished exporting logs for {}", pod_name);
+            file_names.push(file_name);
+            Ok::<(), anyhow::Error>(())
+        }
+        .await
+        {
+            let pod_name = pod.metadata.name.clone().unwrap_or_default();
+            eprintln!("Failed to process pod {}: {}", pod_name, e);
+        }
+    }
+    Ok(file_names)
+}
+
+/**
+ * Write k8s information similar to `kubectl get all` to `mz-k8s-get-all.<namespace>.log`.
+ * Gets output of pods, services, replicasets, deployments, network policies, and statefulsets similar to k9s.
+ * Returns the file name on success.
+ */
+async fn dump_k8s_get_all(client: Client, namespace: &String) -> Result<String, anyhow::Error> {
+    #[derive(Tabled)]
+    struct PodInfo {
+        #[tabled(rename = "NAME")]
+        name: String,
+        #[tabled(rename = "READY")]
+        ready: String,
+        #[tabled(rename = "RESTARTS")]
+        restarts: i32,
+        #[tabled(rename = "STATUS")]
+        status: String,
+        #[tabled(rename = "IP")]
+        ip: String,
+        #[tabled(rename = "NODE")]
+        node: String,
+        #[tabled(rename = "AGE")]
+        age: String,
+    }
+
+    #[derive(Tabled)]
+    struct ServiceInfo {
+        #[tabled(rename = "NAME")]
+        name: String,
+        #[tabled(rename = "TYPE")]
+        type_: String,
+        #[tabled(rename = "CLUSTER-IP")]
+        cluster_ip: String,
+        #[tabled(rename = "EXTERNAL-IP")]
+        external_ip: String,
+        #[tabled(rename = "PORT(S)")]
+        ports: String,
+        #[tabled(rename = "AGE")]
+        age: String,
+    }
```

Comment:
We probably don't need most of this info, but we may want to know if the service has endpoints (ie: is it ready).